pkg/scheduler/core/generic_scheduler.go

```diff
@@ -479,20 +478,22 @@ func (g *genericScheduler) findNodesThatFit(ctx context.Context, state *framewor
 		state.Write(migration.PredicatesStateKey, &migration.PredicatesStateData{Reference: meta})
 
 		checkNode := func(i int) {
-			nodeName := g.cache.NodeTree().Next()
+			nodeInfo := g.nodeInfoSnapshot.NodeInfoList[i]
+			if nodeInfo == nil {
```

Comment:
Can this be nil? Or can we make sure it's not nil?

Reply:
done.